web/ui/react-app/src/pages/flags/Flags.tsx

```diff
@@ -14,18 +17,108 @@ interface FlagsProps {
   data?: FlagMap;
 }
 
+const compareAlphaFn = (keys: boolean, reverse: boolean) => (
+  [k1, v1]: [string, string],
+  [k2, v2]: [string, string]
+): number => {
+  const a = keys ? k1 : v1;
+  const b = keys ? k2 : v2;
+  const reverser = reverse ? -1 : 1;
+  return reverser * a.localeCompare(b);
+};
+
+const boldSubstring = (s: string, subs: string) => {
+  if (subs === '' || s.indexOf(subs) < 0) {
+    return <span>{s}</span>;
+  }
+  return <span dangerouslySetInnerHTML={{ __html: s.split(subs).join(subs.bold()) }} />;
+};
+
+const getSortIcon = (b: boolean | undefined): IconDefinition => {
+  if (b === undefined) {
+    return faSort;
+  }
+  if (b) {
+    return faSortDown;
+  }
+  return faSortUp;
+};
+
+interface ColumnState {
+  name: string;
+  alpha: boolean;
+  focused: boolean;
+}
+
+interface FlagState {
+  search: string;
+  sort: ColumnState;
+}
+
 export const FlagsContent: FC<FlagsProps> = ({ data = {} }) => {
+  const initialState: FlagState = {
```

Comment:
Isn't the idea with hooks to get rid of the single big `state` object per component and just have multiple `useState()` calls for the things you want to update individually? So you would have one for `search` and one for `sort`, and then just call `setSort()` and `setSearch()`.